### PR TITLE
Add LCCC dataset

### PR DESCRIPTION
**Pull request by silverriver**
Hi, I am trying to add a new dataset lccc.
All tests are passed.

### CODE REVIEW

File: datasets/lccc/README.md [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+language_creators:
+- other
+languages:
+- zh
+licenses:
+- mit
+multilinguality:
+- monolingual
+pretty_name: 
+- lccc
```
Comment (albertvillanova) [inline datasets/lccc/README.md:13]: Tjhe value of `pretty_name` should be a string scalar, not a sequence:
```suggestion
pretty_name: LCCC
```

File: datasets/lccc/README.md [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+      - [Who are the annotators?](#who-are-the-annotators)
+    - [Personal and Sensitive Information](#personal-and-sensitive-information)
+  - [Considerations for Using the Data](#considerations-for-using-the-data)
+    - [Social Impact of Dataset](#social-impact-of-dataset)
+    - [Discussion of Biases](#discussion-of-biases)
+    - [Other Known Limitations](#other-known-limitations)
+  - [Additional Information](#additional-information)
+    - [Dataset Curators](#dataset-curators)
+    - [Licensing Information](#licensing-information)
+    - [Citation Information](#citation-information)
```
Comment (albertvillanova) [inline datasets/lccc/README.md:54]: Missing section:
```suggestion
    - [Citation Information](#citation-information)
    - [Contributions](#contributions)
```

File: datasets/lccc/README.md [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+
+## Dataset Description
+
+- **Homepage:** https://github.com/thu-coai/CDial-GPT
```
Comment (albertvillanova) [inline datasets/lccc/README.md:58]: Homepage here should only be given if different from the GitHub repository:
```suggestion
```

File: datasets/lccc/README.md [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+
+LCCC is in Chinese
+
+LCCC中的对话是中文的
+
+## Dataset Structure
+
+### Data Instances
+
+["火锅 我 在 重庆 成都 吃 了 七八 顿 火锅", "哈哈哈哈 ！ 那 我 的 嘴巴 可能 要 烂掉 ！", "不会 的 就是 好 油腻"]
```
Comment (albertvillanova) [inline datasets/lccc/README.md:83]: I think each data instance has a "dialog" key:
```suggestion
{
    "dialog": ["火锅 我 在 重庆 成都 吃 了 七八 顿 火锅", "哈哈哈哈 ！ 那 我 的 嘴巴 可能 要 烂掉 ！", "不会 的 就是 好 油腻"]
}
```
Please put this example between triple backticks.

File: datasets/lccc/README.md [hunk 5]
```diff
@@ -0,0 +1,156 @@
...
+
+### Data Fields
+
+Each line is a list of utterances that consist a dialogue.
+Note that the LCCC dataset provided in our original Github page is in json format,
+however, we are providing LCCC in jsonl format here.
```
Comment (albertvillanova) [inline datasets/lccc/README.md:89]: This section is about the data fields in the generated dataset, not the original source. Maybe something like:

```suggestion
- `dialog` (list of strings): List of utterances consisting of a dialogue.
```

File: datasets/lccc/README.md [hunk 6]
```diff
@@ -0,0 +1,156 @@
...
+
+### Data Splits
+
+We do not provide the offical split for LCCC-large.
+But we provide a split for LCCC-base:
+
+|train|valid|test|
+|:---:|:---:|:---:|
```
Comment (albertvillanova) [inline datasets/lccc/README.md:97]: Better aligned to the right:
```suggestion
|---:|---:|---:|
```

File: datasets/lccc/README.md [hunk 7]
```diff
@@ -0,0 +1,156 @@
...
+
+[Needs More Information]
+
+### Licensing Information
+
+[Needs More Information]
+
+### Citation Information
+
+[Needs More Information]
```
Comment (albertvillanova) [inline datasets/lccc/README.md:156]: Missing information: please add the citation between triple backticks
```suggestion
@inproceedings{wang2020chinese,
  title={A Large-Scale Chinese Short-Text Conversation Dataset},
  author={Wang, Yida and Ke, Pei and Zheng, Yinhe and Huang, Kaili and Jiang, Yong and Zhu, Xiaoyan and Huang, Minlie},
  booktitle={NLPCC},
  year={2020},
  url={https://arxiv.org/abs/2008.03946}
}

### Contributions

Thanks to [@github-username](https://github.com/<github-username>) for adding this dataset.
```
Also please fill the Contributions section with your GitHub username.

**Review by albertvillanova — CHANGES_REQUESTED** [review]
Thanks @silverriver for adding this dataset!

The CI tests are still red: some comment below to fix them. Also, some additional comments so that the dataset card (README file) is more accurate.

File: datasets/lccc/lccc.py [hunk 8]
```diff
@@ -0,0 +1,136 @@
...
+    def _split_generators(self, dl_manager):
+        urls = _URLS[self.config.name]
+        downloaded_data = dl_manager.download_and_extract(urls)
+        if self.config.name == "large":
+            return [
+                datasets.SplitGenerator(
+                    name=datasets.Split.TRAIN,
+                    gen_kwargs={
+                        "filepath": os.path.join(downloaded_data),
+                        "split": "train",
```
Comment (albertvillanova) [inline datasets/lccc/lccc.py:101]: I just realized `split` is not used in `_generate_examples`: therefore, there is no need to pass it. See suggested changes below.
```suggestion
```

File: datasets/lccc/lccc.py [hunk 9]
```diff
@@ -0,0 +1,136 @@
...
+                datasets.SplitGenerator(
+                    name=datasets.Split.TRAIN,
+                    gen_kwargs={
+                        "filepath": os.path.join(downloaded_data["train"]),
+                        "split": "train",
+                    },
+                ),
+                datasets.SplitGenerator(
+                    name=datasets.Split.TEST,
+                    gen_kwargs={"filepath": os.path.join(downloaded_data["test"]), "split": "test"},
```
Comment (albertvillanova) [inline datasets/lccc/lccc.py:116]:
```suggestion
                    gen_kwargs={"filepath": os.path.join(downloaded_data["test"])},
```

File: datasets/lccc/lccc.py [hunk 10]
```diff
@@ -0,0 +1,136 @@
...
+                ),
+                datasets.SplitGenerator(
+                    name=datasets.Split.VALIDATION,
+                    gen_kwargs={
+                        "filepath": os.path.join(downloaded_data["valid"]),
+                        "split": "dev",
```
Comment (albertvillanova) [inline datasets/lccc/lccc.py:122]:
```suggestion
```

File: datasets/lccc/lccc.py [hunk 11]
```diff
@@ -0,0 +1,136 @@
...
+                    },
+                ),
+            ]
+
+    # method parameters are unpacked from `gen_kwargs` as given in `_split_generators`
+    def _generate_examples(self, filepath, split):
```
Comment (albertvillanova) [inline datasets/lccc/lccc.py:128]:
```suggestion
    def _generate_examples(self, filepath):
```

**Comment by silverriver** [comment]
Thank you very much for your help @albertvillanova .

I think have fixed all your comment

File: datasets/lccc/README.md [hunk 12]
```diff
@@ -0,0 +1,167 @@
...
+
+LCCC中的对话是中文的
+
+## Dataset Structure
+
+### Data Instances
+
+{
+    "dialog": ["火锅 我 在 重庆 成都 吃 了 七八 顿 火锅", "哈哈哈哈 ！ 那 我 的 嘴巴 可能 要 烂掉 ！", "不会 的 就是 好 油腻"]
+}
```
[the remark this answers is not in the document]
Comment (silverriver) [reply datasets/lccc/README.md:85]: Done

**Comment by lhoestq** [comment]
Hi ! The dataset seems to be a duplicate of https://huggingface.co/datasets/silver/lccc - next time no need to add it on github if it's already available on huggingface.co ;)

**Comment by silverriver** [comment]
> Hi ! The dataset seems to be a duplicate of https://huggingface.co/datasets/silver/lccc - next time no need to add it on github if it's already available on huggingface.co ;)

OK, sorry for the inconvenience. I have closed another two PRs since these datasets are already available on huggingface.co

**Comment by lhoestq** [comment]
It's fine, thanks @silverriver for adding these datasets !

**Pull request merged**